config/config.py

```diff
@@ -1,108 +1,97 @@
-                               ' on the our team color, removed them from config file')
-        field_port_file = self['COMMUNICATION']['field_port_file']
+        config_parser = ConfigParser(allow_no_value=False)
         try:
-            config_parser.read_file(open(field_port_file))
+            config_parser.read_file(open(filename))
         except FileNotFoundError:
-            print('Impossible de lire le fichier de configuration.\nLoading default simulation with kalman settings')
-            config_parser.read_dict(default_dict)
+            self.logger.critical('The .cfg file {} was not found.'.format(filename))
+            exit(1)
         except ParsingError:
-            print('Le fichier de configuration est mal configuré.\nExiting!')
+            self.logger.critical('The .cfg file {} was not parse correctly.'.format(filename))
             exit(1)
-        field_port = {s: dict(config_parser.items(s)) for s in config_parser.sections()}
-        self['COMMUNICATION'] = field_port['COMMUNICATION']
+
+        config_dict = {section: dict(config_parser.items(section)) for section in config_parser.sections()}
+
+        return config_dict
+
+    def update_content(self):
 
         if 'play_zone' not in self['GAME']:
             self['GAME']['play_zone'] = 'full'
-        if self['GAME']['play_zone'] not in ['full', 'positive', 'negative']:
-            raise RuntimeError('play_zone is either full, positive or negative')
+
+        self['IMAGE']['number_of_camera'] = int(self['IMAGE']['number_of_camera'])
+        self['GAME']['ai_timestamp'] = float(self['GAME']['ai_timestamp'])
 
         # DO NOT TOUCH EVER THEY ARE HARDCODED BOTH IN THE IA AND IN UI-DEBUG
-        our_color = self['GAME']['our_color']
-        if our_color == 'blue':
+        if self['GAME']['our_color'] == 'blue':
             self['COMMUNICATION']['ui_cmd_sender_port'] = 14444    # DO NOT TOUCH
             self['COMMUNICATION']['ui_cmd_receiver_port'] = 15555  # DO NOT TOUCH
-        elif our_color == "yellow":
+        else:
             self['COMMUNICATION']['ui_cmd_sender_port'] = 16666    # DO NOT TOUCH
             self['COMMUNICATION']['ui_cmd_receiver_port'] = 17777  # DO NOT TOUCH
-        else:
-            ValueError("Config file contains wrong colors! Should be either blue or yellow, not {}".format(our_color))
 
-        # [print(key,':' ,value) for key, value in self['COMMUNICATION'].items()]
-        self['IMAGE']['number_of_camera'] = int(self['IMAGE']['number_of_camera'])
-        self['GAME']['ai_timestamp'] = float(self['GAME']['ai_timestamp'])
+    def validate_user_input(self):
 
-        self._config_was_set = True
+        do_exit = False
+        for section, fields in mandatory_fields.items():
+            for field in fields:
+                if field not in self[section]:
+                    self.logger.critical('Mandatory field \'{}\' is missing from section \'{}\''.format(field, section))
+                    do_exit = True
```

Comment:
You could raise runtime error.
exit(1) won't crash the AI, if I remember correctly

Reply:
The config is init before the Framework. I think it should crahs everything.

Reply:
tested, 
It crash :)

Reply:
Peux-tu essayer? Genre mettre un config file qui n'existe pas.

Reply:
ca crash